src/components/register-form.js

```diff
@@ -118,111 +118,24 @@ export class RegisterForm extends React.Component {
-              onChange={this.onChangePassword}
-            />
-          </div>
-          <div className="form-group">
-            <label>Confirm password</label>
-            <input
-              type="password"
-              className={`form-control ${
-                this.isValidPassword() ? 'is-valid' : ''
-                }`}
-              placeholder="Confirm password"
-              value={this.passwordConfirm}
-              onChange={this.onChangeConfirmPassword}
-            />
-          </div>
-
-          <div className="form-group">
-            <label>Select preferences</label>
-            <br />
-            {listOfPreferences.map(preference => (
-              <div className="form-check form-check-inline" key={preference}>
-                <input
-                  className="form-check-input"
-                  type="checkbox"
-                  value={preference}
-                  onChange={e => this.onChangePreference(e, preference)}
-                  checked={
-                    !!this.state.preferences.find(pref => preference === pref)
-                  }
-                />
-                <label className="form-check-label">{preference}</label>
-              </div>
-            ))}
-          </div>
-
-          <div className="form-group">
-            <label>Select age</label>
-            <select
-              onChange={this.onChangeAge}
-              value={this.state.age}
-              className="custom-select"
-            >
-              {this.fillAgesDropdown()}
-            </select>
-          </div>
-
-          <div className="form-group">
-            <label>Select gender</label>
-            <div
-              className="custom-control custom-radio"
-              onClick={e => this.onChangeGender(e, true)}
-            >
-              <input
-                type="radio"
-                className="custom-control-input"
-                checked={this.state.isMale}
-              />
-              <label className="custom-control-label">Male</label>
-            </div>
-            <div
-              className="custom-control custom-radio"
-              onClick={e => this.onChangeGender(e, false)}
-            >
-              <input
-                type="radio"
-                className="custom-control-input"
-                checked={!this.state.isMale}
-              />
-              <label className="custom-control-label">Female</label>
-            </div>
-          </div>
+          <Input type="email" id="email" value={this.state.email} placeholder="Enter email" label="Email address"
+            onChangeInputText={this.onChangeInputText} validation={this.isValidEmail} toggleError={this.state.touched.email} onBlur={this.handleBlur}
+            errorText="Please provide a valid email." />
+          <Input type="password" id="password" value={this.state.password} placeholder="Password" toggleError={this.state.touched.password} label="Password"
```

Comment:
Same comment shared above about legibility of components

Reply:
Fixed, each prop in a single line.